.github/CODEOWNERS

```diff
@@ -127,6 +127,7 @@ extension/sumologicextension/                     @open-telemetry/collector-cont
 internal/aws/                                     @open-telemetry/collector-contrib-approvers @Aneurysm9 @mxiamxia
 internal/collectd/                                @open-telemetry/collector-contrib-approvers @atoulme
 internal/coreinternal/                            @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
+internal/datadog/                                 @open-telemetry/collector-contrib-approvers @IbraheemA @songy23
```

Comment:
We already have pkg/datadog, what is the benefit of having internal/datadog? From what I can tell they are the same